src/components/Endpoint.tsx

```diff
@@ -181,35 +214,39 @@ const Endpoint = (props) => {
     if (props.selectedJupyterItems.directories.length === 0 || props.selectedJupyterItems.directories.length > 1) {
       setLoading(false);
       setAPIError({
-        response: { status: '500', statusText: 'Please select one jupyter directory to transfer data to' },
+        response: {
+          status: '500',
+          statusText: 'To transfer to Jupyter Hub, you must select only one directory to transfer to.',
+        },
       });
-    }
-
-    // Loop through selectedEndpointItems from state
-    for (let selectedEndpointItem of selectedEndpointItems) {
-      transferItems.push({
-        source_path: `${endpointList.path}${selectedEndpointItem.name}`,
-        destination_path: `${config.collection_base_path}/${props.selectedJupyterItems.directories[0].path}/${selectedEndpointItem.name}`,
-        recursive: selectedEndpointItem.type == 'dir' ? true : false,
-      });
-    }
-
-    let transferRequest = {
-      source_endpoint: sourceEndpoint,
-      destination_endpoint: destinationEndpoint,
-      DATA: transferItems,
-    };
+    } else {
+      // Loop through selectedEndpointItems from state
+      for (let selectedEndpointItem of selectedEndpointItems) {
+        transferItems.push({
+          source_path: `${endpointList.path}${selectedEndpointItem.name}`,
+          destination_path: `${config.collection_base_path}/${props.selectedJupyterItems.directories[0].path}/${selectedEndpointItem.name}`,
+          recursive: selectedEndpointItem.type == 'dir' ? true : false,
```

Comment:
I see destination has a path separator between items, but source doesn't. Does it need one?

Reply:
It should not need one. Did you get any errors?

Reply:
I could add some logic to check if a path separator is needed or not.

Reply:
https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L228-L229
https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L284-L288
https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L300-L304